chapters/CRUD-Operations.md

```diff
@@ -358,47 +434,50 @@ reached, or the blockchain history passes the supplied `targetTime`. At this
-   blockchain history. An integer greater of equal to 0. 
-* `currentVersionId`: The version of the contemporaryDIDDocument. An integer starting from 
-   1 and incrementing by 1 with each ::DID Update Payload:: applied to the DID document.
-* `targetVersionId`: The version of the DID document that the resolution algorithm is attempting to resolve.
-* `targetTime`: A 64-bit UNIX timestamp that can be used to target specific historical states of a DID document. 
+   blockchain history. an integer greater of equal to 0; REQUIRED; integer
+* `currentVersionId` - the version of the ::contemporary DID document::. An integer starting from 
+   1 and incrementing by 1 with each ::DID Update Payload:: applied to the DID document; REQUIRED; integer
+* `targetVersionId` - the version of the DID document that the resolution algorithm is attempting to resolve;
+   OPTIONAL; integer
+* `targetTime` - a 64-bit UNIX timestamp that can be used to target specific historical states of a DID document.
    Only ::Beacon Signals:: included in the Bitcoin blockchain before the `targetTime` are processed by the
-   resolution algorithm.
-* `didDocumentHistory`: An ordered array of DID documents from version 1 up to the
-   current version.
-* `updateHashHistory`: An ordered array of SHA256 hashes of ::DID Update Payloads:: 
+   resolution algorithm; REQUIRED; integer
+* `didDocumentHistory` - an ordered array of DID documents from version 1 up to the
+   current version; REQUIRED; array
+* `updateHashHistory` - an ordered array of SHA256 hashes of ::DID Update Payloads:: 
    that have been applied to the DID document by the resolution algorithm in order 
-   to construct the `contemporaryDIDDocument`.
-* `signalsMetadata`: A Map from Bitcoin transaction identifiers of ::Beacon Signals:: 
+   to construct the `contemporaryDIDDocument`; REQUIRED; array
+* `signalsMetadata` - a Map from Bitcoin transaction identifiers of ::Beacon Signals:: 
    to a struct containing ::Sidecar Data:: for that signal provided as part of the
-   resolutionOptions. This struct contains the following properties:
-   * `updatePayload`: A ::DID Update Payload:: which should match the update announced
+   `resolutionOptions`; REQUIRED; map containing the following properties:
+   * `updatePayload` - a ::DID Update Payload:: which SHOULD match the update announced
       by the ::Beacon Signal::. In the case of a ::SMT:: proof of non-inclusion no
-      DID Update Payload may be provided.
-   * `proofs`: A ::Sparse Merkle Tree:: proof that the provided `updatePayload` value is
-      the value at the leaf indexed by the **did:btc1** being resolved. TODO: What exactly this
+      DID Update Payload MAY be provided; OPTIONAL; object
+   * `proofs` - a ::Sparse Merkle Tree:: proof that the provided `updatePayload` value is
+      the value at the leaf indexed by the **did:btc1** being resolved. REQUIRED; TODO: What exactly this
```

Comment:
@KDean-Dolphin can you chime in here. This is defined somewhere now (in your pending PR.)  So happy to get this in as is. But we need to track that this TODO needs fixing asap